hathor/consensus.py

```diff
@@ -312,6 +322,11 @@ def update_voided_info(self, block: Block) -> None:
                 # Either we have a single best chain or all chains have already been voided.
                 assert len(valid_heads) <= 1, 'We must never have more than one valid head'
 
+                # Emit reorg start event, only once
+                if not self.context.reorg_started:
```

Comment:
This method is just called once per block. Shouldn't we assert that `self.context.reorg_started is False`?

Reply:
This is just like the previous one. It makes sense, but I'm not sure if this is a guarantee that we have. Is it not possible to have "nested" reorgs? Externally I can present them as a single reorg, but internally I'm not sure if that's not possible, if it isn't, specially if we have tests that cover that, then I think the assert would be great.

Reply:
Yeah but I misunderstood our implementation, I thought we had a recursive implementation when updating the scores, where it would be possible to temporarily have a "reorg" but after advancing the updates have another "reorg".

Reply:
Done.

Reply:
Great! Thanks!